src/metabase/models/setting.clj

```diff
@@ -1001,6 +1001,10 @@
         (throw (ex-info (tru "Setting {0} allows both user-local and database-local values; this is not supported"
                              setting-name)
                         {:setting setting})))
+      (when (and (some? (:default setting)) (:init setting))
+        (throw (ex-info (tru "Setting {0} uses both :default and :init options, which are mutually exclusive"
+                             setting-name)
+                        {:setting setting})))
```

Comment:
The reason for the conflict was that we hadn't backported this check. There was no harm in adding it.